### PR TITLE
Added AutoIt, language updates and support

### PR DESCRIPTION
**Pull request by Existential-nonce**
## What does this PR do?
Add resources, Improve repo

### Description
- This PR is basically just 2 things at once. I added language support for french and contributed some miscellaneous stuff on the available languages which had outdated hyperlinks to other translation files of the same `.md` file.
- added french translation of CONTRIBUTING.md
> (I admit I kinda abused google translate for the translation but it should be readable and understandable to any french speaking person. I'm aware if I just directly copy pasted everything lazily from google translate then there will be a lot of issues or just generally wrong grammar, so I corrected/guided the faulty translations that were wrong which took me more than an hour.)
- changed small grammatical stuff in the main CONTRIBUTING.md file 
- fixed the markdown section links in the vietnamese CONTRIBUTING-vn.md file
- Included newer language files in the HOWTO.md header files of non-english translations
- added AutoIt language playlist in courses and the docs in their appropriate sections

### Why is this valuable (or not)?
I believe AutoIt is an extremely useful tool and language for windows users, as it can practically do anything when it comes to automation. It can basically do the same things as AutoHotKey but Its syntax is harder to learn (imo) than AutoIt and should serve as an alternative to AHK. 

### How do we know it's really free?
It's the official documentation website for AutoIt. It does not require anything other than an internet connection.

### For book lists, is it a book? For course lists, is it a course? etc.
I added both a course and a book.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/master/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

## Followup

- Check the output of Travis-CI for linter errors!

### CODE REVIEW

**Comment by Existential-nonce** [comment]
Sorry about the weird changes I've made in `free-courses-en.md` and `free-programming-books.md`, I'm not entirely sure how I even managed to change all asterisks into hyphens for 250+ lines. But anyway, the PR should be good now.

**Comment by davorpa** [comment]
Thanks for explain huge changes minutely  :smile:

> Sorry about the weird changes I've made in `free-courses-en.md` and `free-programming-books.md`, I'm not entirely sure how I even managed to change all asterisks into hyphens for 250+ lines. But anyway, the PR should be good now.

I suggest to disable editor linter addons like vscode `markdown-all-in-one`. We use one built on top of @remarkjs.

I see you could solve it in the 5th commit :+1:.

---

If you accept an advice, try to don't overscope PRs :wink:... It makes hard to review it, especially when mixing additions and deletions of if implies many files.

If you want submit more than one at same time, making branches (one branch per PR) instead work on master is the best option. Fetch and checkout them from upstream master.

This has also the advantage of unnecesary extra merge commits/conflicts by your side (e.g. commits 2nd https://github.com/EbookFoundation/free-programming-books/pull/5554/commits/a8f4bffbd521e419d765eb3131cde010097eacf5, 4th https://github.com/EbookFoundation/free-programming-books/pull/5554/commits/6f60e7170389c2d8c4bee34b5a46e78cc5526421) :rocket:

**Comment by Existential-nonce** [comment]
I see, thanks for the explanation. I'll try to be a bit more careful next time.

**Review by davorpa — CHANGES_REQUESTED** [review]
Thanks for complete #5024 and #5454.

Some minor changes :wink: :arrow_down:

Could you find some other interesting AutoIt resources in both listings (books/course)? We try to avoid single items sections especially in these two which are too big (see #5092)

Thanks in advance.

File: courses/free-courses-en.md [hunk 1]
```diff
@@ -125,6 +126,11 @@
 * [Introductory Intel x86: Architecture, Assembly, Applications, & Alliteration](http://opensecuritytraining.info/IntroX86.html)
 
 
+### AutoIt
+
+* [AutoIt Scripting Tutorial](https://www.youtube.com/playlist?list=PL4Jcq5zn02jKpjX0nqI1_fS7mEEb5tw6z) - TutsTeach (HTML)
```
[the remark this answers is not in the document]
Comment (Existential-nonce) [reply courses/free-courses-en.md:131]: Oh yeah my bad, I think I added the HTML indication in the wrong section. I was gonna add it in the courses section. Thanks for pointing it out.

**Comment by eshellman** [comment]
Formidable!

File: HOWTO.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-Read this in other languages: [Español](HOWTO-es.md), [简体中文](HOWTO-zh.md), [हिन्दी](HOWTO-hi.md), [繁體中文](HOWTO-zh-TW.md), [Português (BR)](HOWTO.pt_BR.md), [فارسی](HOWTO-fa_IR.md)
+Read this in other languages: [Français](CONTRIBUTING-fr.md), [Español](HOWTO-es.md), [简体中文](HOWTO-zh.md), [हिन्दी](HOWTO-hi.md), [繁體中文](HOWTO-zh-TW.md), [Português (BR)](HOWTO.pt_BR.md), [فارسی](HOWTO-fa_IR.md)
```
[the remark this answers is not in the document]
Comment (davorpa) [reply HOWTO.md:1]: Fixed by #5556

**Pull request merged**